cmd/otelcorecol/main_windows.go

```diff
@@ -8,41 +8,33 @@ package main
 import (
 	"fmt"
 	"os"
+	"syscall"
 
+	"golang.org/x/sys/windows"
 	"golang.org/x/sys/windows/svc"
 
 	"go.opentelemetry.io/collector/otelcol"
 )
 
 func run(params otelcol.CollectorSettings) error {
-	if useInteractiveMode, err := checkUseInteractiveMode(); err != nil {
-		return err
-	} else if useInteractiveMode {
-		return runInteractive(params)
-	} else {
-		return runService(params)
-	}
-}
-
-func checkUseInteractiveMode() (bool, error) {
-	// If environment variable NO_WINDOWS_SERVICE is set with any value other
-	// than 0, use interactive mode instead of running as a service. This should
-	// be set in case running as a service is not possible or desired even
-	// though the current session is not detected to be interactive
+	// There shouldn't be any reason to use NO_WINDOWS_SERVICE anymore, but,
+	// keeping it as a forcing mechanism or if someone is concerned about
+	// the cost of attempting to run as a service before falling back to
+	// interactive mode.
 	if value, present := os.LookupEnv("NO_WINDOWS_SERVICE"); present && value != "0" {
```

Comment:
Can we deprecate it now if there is nothing other than an extra few milliseconds? Or we are not sure?

Reply:
9434655b9